libraries/botbuilder-core/src/transcriptLogger.ts

```diff
@@ -52,23 +52,23 @@ export class TranscriptLoggerMiddleware implements Middleware {
 
             activities.map((a: Partial<Activity>, index: number) => {
                 const clonedActivity = this.cloneActivity(a);
-                if (index < responses.length) {
-                    if (!clonedActivity.id) {
-                        clonedActivity.id = responses[index].id;
-
-                        // For certain channels, a ResourceResponse with an id is not always sent to the bot.
-                        // This fix uses the timestamp on the activity to populate its id for logging the transcript.
-                        // If there is no outgoing timestamp, the current time for the bot is used for the activity.id.
-                        // See https://github.com/microsoft/botbuilder-js/issues/1122
-                        if (!clonedActivity.id) {
-                            if (clonedActivity.timestamp) {
-                                clonedActivity.id = new Date(clonedActivity.timestamp).getTime().toString();
-                            } else {
-                                clonedActivity.id = Date.now().toString();
-                            }
-                        }
+                if (index < responses.length && !clonedActivity.id) {
```

Comment:
We should always set the resourceResponse Id on the activity to be logged down here.

Reply:
Good point ... removed '!clonedActivity.id'